rustls/src/rand.rs

```diff
@@ -1,25 +1,26 @@
 use crate::msgs::codec;
+use crate::TLSError;
 /// The single place where we generate random material
 /// for our own use.  These functions never fail,
 /// they panic on error.
 use ring::rand::{SecureRandom, SystemRandom};
 
 /// Fill the whole slice with random material.
-pub fn fill_random(bytes: &mut [u8]) {
-    SystemRandom::new().fill(bytes).unwrap();
+pub fn fill_random(bytes: &mut [u8]) -> Result<(), TLSError> {
```

Comment:
Can we have a separate `GetRandomFailed` error type?  This can avoid size of return value too large.